File: lsp-mcp-bridge/testutil/fixtures/sample.go
// Acceptance test fixture — Go.
//
// Known facts for test assertions:
//   - NewWorker() return type : *Worker
//   - Worker.Run() is defined at the line marked DEFINITION
//   - Run() is called at the line marked REFERENCE
package sample

import "fmt"

// Worker processes jobs identified by an integer ID.
type Worker struct {
	ID int
}

// Run executes the worker and returns a status string. // DEFINITION
func (w *Worker) Run() string {
	return fmt.Sprintf("worker-%d", w.ID)
}

// NewWorker constructs a Worker with the given ID.
func NewWorker(id int) *Worker {
	return &Worker{ID: id}
}

// Dispatch creates a worker and runs it, returning the status.
func Dispatch(id int) string {
	w := NewWorker(id)
	return w.Run() // REFERENCE
}

File: lsp-mcp-bridge/testutil/fixtures/sample.py
"""
Acceptance test fixture — Python.

Known facts for test assertions:
  - get_result() return type : Future[tuple[bytes, int]]
  - process() parameter `fut`: Future[tuple[bytes, int]]
  - Worker.submit() is defined at the line marked DEFINITION
  - submit() is called at the line marked REFERENCE
"""
from __future__ import annotations

from concurrent.futures import Future


class Worker:
    def submit(self, payload: bytes) -> Future[tuple[bytes, int]]:  # DEFINITION
        """Submit payload and return a future resolving to (result, code)."""
        raise NotImplementedError


def get_result() -> Future[tuple[bytes, int]]:
    w = Worker()
    return w.submit(b"ping")  # REFERENCE


def process(fut: Future[tuple[bytes, int]]) -> bytes:
    result = fut.result()  # hover here → Future[tuple[bytes, int]]
    value, _code = result
    return value
